3-UsingAPIs/Week2/homework/ex2-pokemonApp/index.js

```diff
@@ -22,18 +23,83 @@ Use async/await and try/catch to handle promises.
 Try and avoid using global variables. As much as possible, try and use function 
 parameters and return values to pass data back and forth.
 ------------------------------------------------------------------------------*/
-function fetchData(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+function createElements() {
+  const button = document.createElement('button');
+  button.className = 'button';
+  button.textContent = 'Show Pokemons';
+  button.type = 'button';
+  document.body.appendChild(button);
+
+  const selectElement = document.createElement('select');
+  selectElement.className = 'select-pokemon';
+  document.body.appendChild(selectElement);
+
+  const image = document.createElement('img');
+  image.className = 'pokemon-image';
+  image.src = 'https://';
+  image.alt = '';
+  document.body.appendChild(image);
+}
+
+function addOptions(pokemonsData) {
+  const selectElement = document.querySelector('.select-pokemon');
+
+  pokemonsData.forEach((pokemon) => {
+    const option = document.createElement('option');
+    option.textContent = pokemon.name;
+    option.value = pokemon.url;
+    selectElement.appendChild(option);
+  });
 }
 
-function fetchAndPopulatePokemons(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+async function fetchData(url) {
+  try {
+    const responseData = await fetch(url);
+    const jsonData = await responseData.json();
+    return jsonData;
+  } catch (error) {
+    console.log(error.message);
+  }
 }
 
-function fetchImage(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+async function fetchAndPopulatePokemons(url) {
+  const button = document.querySelector('.button');
+  const selectElement = document.querySelector('.select-pokemon');
+
+  try {
+    const pokemonsData = await fetchData(url);
+    button.addEventListener('click', () => {
+      addOptions(pokemonsData.results);
+
+      selectElement.addEventListener('change', (event) => {
+        fetchImage(event);
+      });
+    });
+  } catch (error) {
+    console.log(error.message);
+  }
+}
+
+async function fetchImage(event) {
+  const pokemonImage = document.querySelector('.pokemon-image');
+  const selectedPokemonURL = event.target.value;
+
+  try {
+    const selectedPokemonData = await fetchData(selectedPokemonURL);
+
+    pokemonImage.src = selectedPokemonData.sprites.front_default;
+    pokemonImage.alt = `Pokemon ${selectedPokemonData.name} Image`;
+  } catch (error) {
+    console.log(error.message);
+  }
 }
 
 function main() {
-  // TODO complete this function
+  const url = 'https://pokeapi.co/api/v2/pokemon?limit=151';
+
+  createElements();
+
+  fetchAndPopulatePokemons(url);
 }
+
+window.addEventListener('load', main());
```

Comment:
You should not call the `main` function when you add it as an event listener. Instead, you should _**pass**_ the function as a parameter.

```suggestion
window.addEventListener('load', main);
```